src/shared/lib/ExtendedRouterStore.ts

```diff
@@ -90,6 +72,13 @@ export default class ExtendedRouterStore extends RouterStore {
             newQuery = newParams;
         }
 
+        // clear out any undefined props
```

Comment:
newQuery will have ALL params in URL, so here we can just delete entrees that are empty. nothing can survive.